examples/timeseries/timeseries_weather_forecasting.py

```diff
@@ -0,0 +1,429 @@
+    "Water vapor concentration",
+    "Airtight",
+    "Wind speed",
+    "Maximum wind speed",
+    "Wind direction in degrees",
+]
+
+feature_keys = [
+    "p (mbar)",
+    "T (degC)",
+    "Tpot (K)",
+    "Tdew (degC)",
+    "rh (%)",
+    "VPmax (mbar)",
+    "VPact (mbar)",
+    "VPdef (mbar)",
+    "sh (g/kg)",
+    "H2OC (mmol/mol)",
+    "rho (g/m**3)",
+    "wv (m/s)",
+    "max. wv (m/s)",
+    "wd (deg)",
+]
+
+colors = [
+    "blue",
+    "orange",
+    "green",
+    "red",
+    "purple",
+    "brown",
+    "pink",
+    "gray",
+    "olive",
+    "cyan",
+]
+
+date_time_key = "Date Time"
+
+
+def show_raw_visualization(data):
+    time_data = data[date_time_key]
+    fig, axes = plt.subplots(
+        nrows=7, ncols=2, figsize=(15, 20), dpi=80, facecolor="w", edgecolor="k"
+    )
+    for i in range(len(feature_keys)):
+        key = feature_keys[i]
+        c = colors[i % (len(colors))]
+        t_data = data[key]
+        t_data.index = time_data
+        t_data.head()
+        ax = t_data.plot(
+            ax=axes[i // 2, i % 2],
+            color=c,
+            title="{} - {}".format(titles[i], key),
+            rot=25,
+        )
+        ax.legend([titles[i]])
+    plt.tight_layout()
+
+
+show_raw_visualization(df)
+
+"""
+This heat map shows the correlation between different features.
+"""
+
+
+def show_heatmap(data):
+    plt.matshow(data.corr())
+    plt.xticks(range(data.shape[1]), data.columns, fontsize=14, rotation=90)
+    plt.gca().xaxis.tick_bottom()
+    plt.yticks(range(data.shape[1]), data.columns, fontsize=14)
+
+    cb = plt.colorbar()
+    cb.ax.tick_params(labelsize=14)
+    plt.title("Feature Correlation Heatmap", fontsize=14)
+    plt.show()
+
+
+show_heatmap(df)
+
+"""
+The input data will include pressure, temperature (in Celsius) and specific humidity.
+The below line graphs show each feature graphed by hour, month and year.
+"""
+
+
+def show_time_based_visualizations(data, idx):
+    selected_feature = feature_keys[idx]
+    fig, axes = plt.subplots(
+        nrows=1, ncols=3, figsize=(15, 5), dpi=80, facecolor="w", edgecolor="k"
+    )
+
+    dataframe = data.copy()
+    dataframe.index = pd.to_datetime(data[date_time_key])
+
+    x_data = [dataframe.index.hour, dataframe.index.month, dataframe.index.year]
+    x_labels = ["Hour of Day", "Month", "Year"]
+
+    for i in range(3):
+        selected_ax = axes[i]
+        sns.lineplot(x=x_data[i], y=dataframe[selected_feature], ax=selected_ax)
+        selected_ax.set(xlabel=x_labels[i], ylabel=titles[idx])
+    plt.tight_layout()
+    plt.show()
+    return
+
+
+# plot pressure
+show_time_based_visualizations(df, 0)
+# plot temperature
+show_time_based_visualizations(df, 1)
+# plot specific humidity
+show_time_based_visualizations(df, 8)
+
+"""
+## Data Preprocessing
+
+Here we are picking 300000 data points for training. The observation is recorded every
+10 mins, that means 6 times in an hour. Thus STEP is equal to 6. We are tracking data
```

Comment:
"Thus STEP is equal to 6." -> explain this when you say "Variable step defines this sampling.", not here